integrators/ordinary_differential_equations.hpp

```diff
@@ -34,9 +34,137 @@ using base::Status;
 using geometry::Instant;
 using numerics::DoublePrecision;
 using quantities::Difference;
+using quantities::Quotient;
 using quantities::Time;
 using quantities::Variation;
 
+// A differential equation of the form y′ = f(y, t).
+// |State| is the type of y.
+template<typename... StateTypes>
+struct ExplicitFirstOrderOrdinaryDifferentialEquation final {
+  using State = std::tuple<std::vector<StateTypes>...>;
+  // NOTE(eggrobin): we cannot use |Variation| here because of a bug in MSVC
+  // when parameter packs interact with defaulted template parameters (and
+  // possibly decltype), but this is just
+  //   std::tuple<std::vector<Variation<StateTypes>>...>.
+  using StateVariation = std::tuple<
+      std::vector<Quotient<Difference<StateTypes, StateTypes>, Time>>...>;
+
+  using RightHandSideComputation =
+      std::function<Status(Instant const& t,
+                           State const& state,
+                           StateVariation& derivatives)>;
+
+  struct SystemState final {
+    SystemState() = default;
+    SystemState(State const& y, Instant const& t);
+
+    std::tuple<std::vector<DoublePrecision<StateTypes>>...> y;
+    DoublePrecision<Instant> time;
+
+    friend bool operator==(SystemState const& lhs, SystemState const& rhs) {
+      return lhs.y == rhs.y && lhs.time == rhs.time;
+    }
+  };
+
+  // We cannot use |Difference<StateTypes>| here for the same reason.  For some
+  // reason |DoublePrecision<StateTypes>| above works...
+  using SystemStateError =
+      std::tuple<std::vector<Difference<StateTypes, StateTypes>>...>;
+
+  // A functor that computes f(y, t) and stores it in |derivatives|.
+  // This functor must be called with |std::get<i>(derivatives).size()| equal to
+  // |std::get<i>(state).size()| for all i, but there is no requirement on the
+  // values in |derivatives|.
```

Comment:
Move this comment to the type declaration.

Reply:
No, `Splitting` needs to have the comment on the member variable declarations because of the difference between `left_flow` and `right_flow`, let's keep this consistent.